src/handler/get_xpub.c

```diff
@@ -3,11 +3,7 @@
 #include <stddef.h>   // size_t
 #include <string.h>   // memset, explicit_bzero
 
-#ifndef TEST
```

Comment:
Tests build successfully without this `#ifndef TEST`, so they were removed

Reply:
The tests run as intended without the `ifndef`, I think it's best to remove them unless the `stubs.h` is supposed to add some functionality on the tests.
This way we don't have to duplicate this on every file that uses an sdk header.

Reply:
I guess this is fine. I still don't fully understand it but I'd really prefer if these small changes that don't affect the feature being implemented to be on a separate PR. These changes are already quite large already.